engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/RequestResponseAkkaHttpHandler.scala

```diff
@@ -0,0 +1,98 @@
+import akka.http.scaladsl.server.directives.DebuggingDirectives
+import akka.stream.Materializer
+import cats.data.{NonEmptyList, Validated}
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.{Encoder, Json}
+import io.circe.generic.JsonCodec
+import io.circe.syntax.EncoderOps
+import pl.touk.nussknacker.engine.api.Context
+import pl.touk.nussknacker.engine.api.exception.NuExceptionInfo
+import pl.touk.nussknacker.engine.requestresponse.DefaultResponseEncoder
+import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter.InterpreterType
+import pl.touk.nussknacker.engine.requestresponse.RequestResponseEngine.RequestResponseResultType
+import pl.touk.nussknacker.engine.requestresponse.api.{RequestResponseGetSource, RequestResponsePostSource}
+import pl.touk.nussknacker.engine.requestresponse.metrics.InvocationMetrics
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class ProcessRoute(processInterpreters: scala.collection.Map[String, InterpreterType]) extends Directives with LazyLogging {
+
+  protected def logDirective(scenarioName: String): Directive0 = DebuggingDirectives.logRequestResult((s"request-response-$scenarioName", Logging.DebugLevel))
+
+  def route(implicit ec: ExecutionContext, mat: Materializer): Route =
+    path(Segment) { processPath =>
+      logDirective(processPath) {
+        processInterpreters.get(processPath) match {
+          case None =>
+            complete {
+              HttpResponse(status = StatusCodes.NotFound)
+            }
+          case Some(processInterpreter) => new RequestResponseAkkaHttpHandler(processInterpreter).invoke {
+            case Invalid(errors) => complete {
+              logErrors(processPath, errors)
+              HttpResponse(status = StatusCodes.InternalServerError, entity = toEntity(errors.toList.map(info => EspError(info.nodeComponentInfo.map(_.nodeId), Option(info.throwable.getMessage)))))
+            }
+            case Valid(results) => complete {
+              HttpResponse(status = StatusCodes.OK, entity = toEntity(results))
+            }
+          }
+        }
+      }
+      //TODO place openApi endpoint
+    } ~ pathEndOrSingleSlash {
+      //healthcheck endpoint
+      get {
+        complete {
+          HttpResponse(status = StatusCodes.OK)
+        }
+      }
+    }
+
+  private def toEntity[T: Encoder](value: T): ResponseEntity = HttpEntity(contentType = `application/json`, string = value.asJson.noSpacesSortKeys)
+
+  private def logErrors(processPath: String, errors: NonEmptyList[NuExceptionInfo[_ <: Throwable]]): Unit = {
+    logger.warn(s"Failed to invoke: $processPath with errors: ${errors.map(_.throwable.getMessage)}")
+    errors.toList.foreach { error =>
+      logger.info(s"Invocation failed $processPath, error in ${error.nodeComponentInfo.map(_.nodeId)}: ${error.throwable.getMessage}", error.throwable)
```

Comment:
why it is info? and why it is double logging here?

Reply:
I lowered levels, intention is a bit similar to log with trace, but we want to log all errors

Reply:
I understand intention but don't understand why you lowered levels. If someone redirect all warn and errors to some file, will not see this messages. It is ok?

Reply:
I think it should be ok. This is not necessarily an error from NU POV - it can be e.g. problem with other services, or wrong data.

Reply:
ok